samples/guestbook/gcp/deploy.go

```diff
@@ -0,0 +1,159 @@
+		fmt.Fprintln(os.Stderr, "deploy:", err)
+		os.Exit(1)
+	}
+}
+
+func deploy(guestbookDir string) error {
+	tfStatePath := filepath.Join(guestbookDir, "gcp", "terraform.tfstate")
+	keys := []string{"project", "cluster_name", "cluster_zone", "bucket", "database_instance", "database_region", "motd_var_config", "motd_var_name"}
+	tfState := make(map[string]string)
+	for _, k := range keys {
+		cmd := exec.Command("terraform", "output", "-state", tfStatePath, k)
+		outb, err := cmd.CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("running %v: %v: %s", cmd.Args, err, outb)
+		}
+		tfState[k] = string(outb)
+	}
+
+	gcp := gcloud{project: tfState["project"]}
+	tempDir, err := ioutil.TempDir("", "guestbook-k8s-")
+	if err != nil {
+		return fmt.Errorf("making temp dir: %v", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Fill in Kubernetes template parameters.
+	proj := strings.Replace(tfState["project"], ":", "/", -1)
+	imageName := fmt.Sprintf("gcr.io/%s/guestbook", proj)
+	gbyin, err := ioutil.ReadFile(filepath.Join(guestbookDir, "gcp", "guestbook.yaml.in"))
+	if err != nil {
+		return fmt.Errorf("reading guestbook.yaml.in: %v", err)
+	}
+	gby := string(gbyin)
+	replacements := map[string]string{
+		"{{IMAGE}}":             imageName,
+		"{{bucket}}":            tfState["bucket"],
+		"{{database_instance}}": tfState["database_instance"],
+		"{{database_region}}":   tfState["database_region"],
+		"{{motd_var_config}}":   tfState["motd_var_config"],
+		"{{motd_var_name}}":     tfState["motd_var_name"],
+	}
+	for old, new := range replacements {
+		gby = strings.Replace(gby, old, new, -1)
+	}
+	if err := ioutil.WriteFile(filepath.Join(tempDir, "guestbook.yaml"), []byte(gby), 0666); err != nil {
+		return fmt.Errorf("writing guestbook.yaml: %v", err)
+	}
+
+	// Build Guestbook Docker image.
+	log.Printf("Building %s...", imageName)
+	build := exec.Command("vgo", "build", "-o", "gcp/guestbook")
+	build.Env = []string{"GOOS=linux", "GOARCH=amd64"}
+	build.Dir = guestbookDir
+	if out, err := build.CombinedOutput(); err != nil {
+		return fmt.Errorf("building guestbook app by running %v: %v: %s", build.Args, err, out)
+	}
+	cmd := gcp.cmd("container", "builds", "submit", "-t", imageName, filepath.Join(guestbookDir, "gcp"))
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("building container image with %v: %v: %s", cmd.Args, err, out)
+	}
+
+	// Run on Kubernetes.
+	log.Printf("Deploying to %s...", tfState["cluster_name"])
+	cmd = gcp.cmd("container", "clusters", "get-credentials", "--zone", tfState["cluster_zone"], tfState["cluster_name"])
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("getting credentials with %v: %v: %s", cmd.Args, err, out)
+	}
+	kubeCmds := [][]string{
+		{"kubectl", "apply", "-f", filepath.Join(tempDir, "guestbook.yaml")},
+		// Force pull the latest image.
+		{"kubectl", "scale", "--replicas", "0", filepath.Join("deployment", guestbookDir)},
```

Comment:
The argument must be the literal string `"deployment/guestbook"`. It's a Kubernetes identifier, not a file path.